AP-protocol.md

```diff
@@ -106,54 +106,73 @@ When the network is first initialized, nodes only know who their own neighbors a
+
+### **Replies, merging and termination condition**
+
+When a **pending** neighbor returns a `FloodData` message, the message is temporarily saved and the node is marked as **done**. When all neighbors are **done** or **failed**, or if there are none (in the case of a node connected to only one neighbor, from which it has received the request) the Node creates its own `FloodData`:
 ```rust
-struct FloodResponse {
-	flood_id: u64,
-	source_routing_header: SourceRoutingHeader,
-	path_trace: Vec<(NodeId, NodeType)>
+pub struct FloodData {
+	// The original flood ID
+    pub flood_id: u64,
+	// Its own node ID
+	pub from:  NodeId
+	// Its own node ID and type
+    pub node_list: HashSet<(NodeId, NodeType)>,
+	// See below for explaination
+    pub connections: HashSet<(NodeId, NodeId)>,
+	// The Node ID of one of the nodes marked as "failed", otherwise None
+    pub error_at: Option<NodeId>,
+	// Kept at "false" 
+    pub is_broadcast: bool 
 }
 ```
+The `connections` field gets populated with the neighbors, creating a tuple of two node IDs, one being the current and one being the neighbor, which MUST be in ascending order: (n1,n2) where n1 < n2 always.
+The node then proceeds to merge all obtained responses, and MUST send the final response to the previous hop it had stored, changing the `from` field to its own ID.
 
-### **Recording Topology Information**
-
-For every flood response or acknowledgment the initiator receives, it updates its understanding of the graph:
-
-- If the node receives a flood response with a **path trace**, it records the paths between nodes. The initiator learns not only the immediate neighbors but also the connections between nodes further out.
-- Over time, as the query continues to flood, the initiator accumulates more information and can eventually reconstruct the entire graph's topology.
+### **Special considerations for Requests containing a "to" field**
+In this case, when returning a `FloodData` packet, the fields `connections` and `node_list` get set to empty, unless:
+	- The node is the target node, in which case it MUST populate `node_list` with itself, and MUST add the connection to the previous hop on the request to `connections`, or
+	- The node is not the target node but has received a non-empty `FloodData`, in which case it MUST add the connection to the previous hop as described above, and CAN add itself to the `node_list`.
 
-### **Termination Condition**
+### **Broadcast messages**
 
-The flood can terminate when:
+When a drone first connects to the network, it creates a `FloodData` message, containing a random `flood_id`, itself as the `node_list` and its neighbors as `connections`, with `is_broadcast` set to true. This message SHOULD get forwarded to neighbouring nodes. When receiving such a message a node MUST forward it to every other neighbor except the one it received it from. If received a second time the message MUST get ignored.
```

Comment:
I like this, it greatly simplifies flooding management, however in theory drones should not be able to initiate flooding. Please correct me if I am wrong

Reply:
That can be discussed at the next WG I guess, it would make it easier for clients and servers to respond to new drone additions without requiring another flood of the whole network. (This part of the proposal isn't mandatory, so it can be skipped if we don't agree on it)